# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/common/string_utils.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/common/string_utils.h
--- a/gfx/angle/checkout/src/common/string_utils.h	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/common/string_utils.h	Fri Nov 08 17:06:00 2019 +0000
@@ -41,12 +41,14 @@
 
 std::string TrimString(const std::string &input, const std::string &trimChars);
 
+// Return the substring starting at offset and up to the first occurance of the |delimeter|.
+std::string GetPrefix(const std::string &input, size_t offset, const char *delimiter);
+std::string GetPrefix(const std::string &input, size_t offset, char delimiter);
+
 bool HexStringToUInt(const std::string &input, unsigned int *uintOut);
 
 bool ReadFileToString(const std::string &path, std::string *stringOut);
 
-Optional<std::vector<wchar_t>> WidenString(size_t length, const char *cString);
-
 // Check if the string str begins with the given prefix.
 // The comparison is case sensitive.
 bool BeginsWith(const std::string &str, const std::string &prefix);